modules/articles/client/services/articles.client.service.js

```diff
@@ -8,12 +8,29 @@
   ArticlesService.$inject = ['$resource'];
 
   function ArticlesService($resource) {
-    return $resource('api/articles/:articleId', {
+    var Article = $resource('api/articles/:articleId', {
       articleId: '@_id'
     }, {
       update: {
         method: 'PUT'
       }
     });
+
+    angular.extend(Article.prototype, {
+      createOrUpdate: function () {
+        var article = this;
+        return createOrUpdate(article);
+      }
+    });
+
+    return Article;
+
+    function createOrUpdate(article) {
```

Comment:
I would just consider moving this above the return, some people might get confused and forget about hoisting.